src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -209,9 +213,14 @@ private String getCacheKey(String url, Multimap<String, String> queryParams) thr
             builder.append("_").append(restoreFactory.getAsUsername());
         }
         builder.append("_").append(uri);
-        for (String key : queryParams.keySet()) {
+        List<String> keyList = queryParams.keySet().stream().collect(Collectors.toList());
+        Collections.sort(keyList);
+        for (String key : keyList) {
```

Comment:
Looking at this again now I think you can simplify this a bit by using a TreeMap which stores it's data in sorted order:
```suggestion
        // Convert Multimap to TreeMap to sort keys
        Map<String, Collection<String>> sortedQueryParams = new TreeMap<>(queryParams.asMap());
        for (String key : sortedQueryParams.keySet()) {
```